chia/_tests/core/full_node/test_address_manager.py

```diff
@@ -586,3 +664,98 @@ async def test_cleanup(self):
+            count_ids: int = 0
+
+            metadata.append(("key", str(address_manager.key)))
+
+            for node_id, info in address_manager.map_info.items():
+                unique_ids[node_id] = count_ids
+                if info.ref_count > 0:
+                    assert count_ids != address_manager.new_count
+                    nodes.append((count_ids, info))
+                    count_ids += 1
+            metadata.append(("new_count", str(count_ids)))
+
+            tried_ids = 0
+            for node_id, info in address_manager.map_info.items():
+                if info.is_tried:
+                    assert info is not None
+                    assert tried_ids != address_manager.tried_count
+                    nodes.append((count_ids, info))
+                    count_ids += 1
+                    tried_ids += 1
+            metadata.append(("tried_count", str(tried_ids)))
+
+            for bucket in range(NEW_BUCKET_COUNT):
+                for i in range(BUCKET_SIZE):
+                    if address_manager.new_matrix[bucket][i] != -1:
+                        index = unique_ids[address_manager.new_matrix[bucket][i]]
+                        new_table_entries.append((index, bucket))
+
+            # Ensure the parent directory exists
+            peers_file_path.parent.mkdir(parents=True, exist_ok=True)
+
+            transformed_nodes: list[tuple[uint64, str]] = []
+            transformed_new_table: list[tuple[uint64, uint64]] = []
+
+            # these lines used to be in another function that was only called as part of the serialization
+            # I've inlined them here for clarity
+
+            for index, [node_id, peer_info] in enumerate(nodes):
+                transformed_nodes.append((uint64(node_id), peer_info.to_string()))
+
+            for index, [node_id, bucket_id] in enumerate(new_table_entries):
+                transformed_new_table.append((uint64(node_id), uint64(bucket_id)))
+
+            # this was also previously its own function, now inlined
+            serialized_bytes: bytes = bytes(PeerDataSerialization(metadata, transformed_nodes, transformed_new_table))
+            await write_file_async(peers_file_path, serialized_bytes, file_mode=0o644)
+
+        # create a file with the old serialization, then migrate to new serialization
+        addrman = AddressManagerTest()
+        now = int(math.floor(time.time()))
+        t_peer1 = TimestampedPeerInfo("250.7.1.1", uint16(8333), uint64(now - 10000))
+        t_peer2 = TimestampedPeerInfo("1050:0000:0000:0000:0005:0600:300c:326b", uint16(9999), uint64(now - 20000))
+        t_peer3 = TimestampedPeerInfo("250.7.3.3", uint16(9999), uint64(now - 30000))
+        source = PeerInfo("252.5.1.1", uint16(8333))
+        await addrman.add_to_new_table([t_peer1, t_peer2, t_peer3], source)
+        await addrman.mark_good(PeerInfo("250.7.1.1", uint16(8333)))
+
+        peers_dat_filename = tmp_path / "peers.dat"
+        if peers_dat_filename.exists():
```

Comment:
this looks like it should also use the `Suppress` context manager

Reply:
done